Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -2115,6 +2116,32 @@ void WebProcessPool::updateAudibleMediaAssertions()
     };
 }
 
+WebProcessWithMediaStreamingToken WebProcessPool::webProcessWithMediaStreamingToken() const
+{
+    return m_webProcessWithMediaStreamingCounter.count();
+}
+
+void WebProcessPool::updateMediaStreamingActivity()
+{
+    if (!m_webProcessWithMediaStreamingCounter.value()) {
+        WEBPROCESSPOOL_RELEASE_LOG(ProcessSuspension, "updateMediaStreamingActivity: The number of processes with media networking now zero. Notify network.");
+        m_mediaStreamingActivity = false;
+#if PLATFORM(COCOA)
```

Comment:
I would be tempted to make the whole method #if PLATFORM(COCOA) or not at all (and make notifyMediaStreamingActivity a no-op in non cocoa ports.

Reply:
I thought about it; but it makes the change a little bit more "ugly" in my opinion as you would need to have `updateMediaStreamingActivity` present in all WebProcessPool implementation, and then `m_mediaStreamingActivity` be COCOA conditional.
Having `updateMediaStreamingActivity` virtual would have been more C++ish, but it would be a bigger change.

So ultimately, I think it's the best I can do without changing the structure too much.